projects/step-core/src/lib/client/augmented/services/augmented-resources-service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { ResourcesService } from '../../generated';
+import { TableRestService } from '../../table/services/table-rest.service';
+import { BaseHttpRequest } from '../../generated/core/BaseHttpRequest';
+import { TableRemoteDataSource } from '../../../modules/table/shared/table-remote-data-source';
+import { Observable } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class AugmentedResourcesService extends ResourcesService {
+  private readonly RESOURCES_TABLE_ID = 'resources';
+
+  private readonly dataSource = new TableRemoteDataSource(this.RESOURCES_TABLE_ID, this._tableRest, {
+    name: 'attributes.name',
+    resourceType: 'resourceType',
+    id: 'id',
+  });
+
+  constructor(override httpRequest: BaseHttpRequest, private _tableRest: TableRestService) {
+    super(httpRequest);
+  }
+
+  public getResourcesTableDataSource(): TableRemoteDataSource<any> {
```

Comment:
@chillbirdo  For typescript the correct variant would be
```
private  _dataSource = new TableRemoteDataSource(...);

get dataSource(): TableRemoteDataSource {
  return this._dataSource;
}
```

So without set accessor we tell, that dataSource can not be redefined anywhere else.
But the shorter variant would be just have a public readonly field
```
readonly dataSource = new TableRemoteDataSource(...);
```

Probably it's another for discussion.
I understand that you follow the Java approach where field's assessors are made with getField/setField methods. I don't think that it make sense to follow the same approach for Typescript